File: STRIVER/video/vid-5/pattern19/p19.py
# ********
# ***  ***
# **    **
# *      *
# *      *
# **    **
# ***  ***
# ********

# formula: 
# space (row-i)
# stuff (2*i)
# space (row-i)

def p19(row:int)->None:

    # upper part 
    for i in range(row):
        # space
        for j in range(row-i):
            print("*",end="")

        # stuff
        for j in range((2*i)):
            print(" ",end="")

        # space
        for j in range(row-i):
            print("*",end="")

        print(" ")

    # lower part 
    for i in range(row-1,-1,-1):
        # space
        for j in range(row-i):
            print("*",end="")

        # stuff
        for j in range((2*i)):
            print(" ",end="")

        # space
        for j in range(row-i):
            print("*",end="")

        print(" ")

row = 5
p19(row)

File: STRIVER/video/vid-5/pattern19/p19.cpp
// ********
// ***  ***
// **    **
// *      *
// *      *
// **    **
// ***  ***
// ********

// formula: 
// space (row-i)
// stuff (2*i)
// space (row-i)


#include <iostream>
using namespace std;

int main()
{
    int row = 5;

    //upper part 
    for (char i = 0; i < row; i++){
        // space
        for(int j = 0; j< row-i; j++){
            cout<<"*";
        }
        // stuff
        for(int j = 0; j< (2*i); j++){
            cout<<"-";
        }

        // space
        for(int j = 0; j< row-i; j++){
            cout<<"*";
        }

        cout<<endl;
    }

    //lower part 
    for (char i = row; i >= 0; i--){
        // space
        for(int j = 0; j< row-i; j++){
            cout<<"*";
        }
        // stuff
        for(int j = 0; j< (2*i); j++){
            cout<<"-";
        }

        // space
        for(int j = 0; j< row-i; j++){
            cout<<"*";
        }

        cout<<endl;
    }

    return 0;
}